examples/LayoutSpecExamples-Swift/Sample/LayoutExampleNode.swift

```diff
@@ -143,3 +143,144 @@ class FlexibleSeparatorSurroundingContent : LayoutExampleNode {
+    static let icon = UIColor.red
+  }
+  
+  required init() {
+    super.init()
+    
+    let avatarImage = UIImage.draw(size: ImageSize.avatar, fillColor: ImageColor.avatar) { () -> UIBezierPath in
+      return UIBezierPath(roundedRect: CGRect(origin: CGPoint.zero, size: ImageSize.avatar), cornerRadius: ImageSize.avatar.width / 20)
+    }
+    
+    let iconImage = UIImage.draw(size: ImageSize.icon, fillColor: ImageColor.icon) { () -> UIBezierPath in
+      return UIBezierPath(ovalIn: CGRect(origin: CGPoint.zero, size: ImageSize.icon))
+    }
+    
+    photoNode1.image = avatarImage
+    photoNode2.image = avatarImage
+    dotNode.image = iconImage
+    
+    badgeNode.attributedText = NSAttributedString.attributedString(string: " 999+ ", fontSize: 20, color: .white)
+    badgeNode.backgroundColor = .red
+    badgeNode.cornerRadius = 12
+    badgeNode.clipsToBounds = true
+  }
+  
+  override func layoutSpecThatFits(_ constrainedSize: ASSizeRange) -> ASLayoutSpec {
+    photoNode.style.preferredSize = ImageSize.avatar
+    iconNode.style.preferredSize = ImageSize.icon
+    
+    let cornerSpec1 = ASCornerLayoutSpec(child: photoNode1, corner: dotNode, location: .topRight)
+    let cornerSpec2 = ASCornerLayoutSpec(child: photoNode2, corner: badgeNode, location: .topRight)
+    let cornerSpec3 = ASCornerLayoutSpec(child: photoNode, corner: iconNode, location: .topRight)
+    
+    cornerSpec1.offset = CGPoint(x: -3, y: 3)
+
+    let stackSpec = ASStackLayoutSpec.vertical()
+    stackSpec.spacing = 40
+    stackSpec.children = [cornerSpec1, cornerSpec2, cornerSpec3]
+    
+    return stackSpec
+  }
+  
+  override class func title() -> String {
+    return "Declarative way for Corner image Layout"
+  }
+  
+  override class func descriptionTitle() -> String? {
+    return nil
+  }
+}
+
+class UserProfileSample : LayoutExampleNode {
+  
+  let badgeNode = ASImageNode()
+  let avatarNode = ASImageNode()
+  let usernameNode = ASTextNode()
+  let subtitleNode = ASTextNode()
+  
+  struct ImageSize {
+    static let avatar = CGSize(width: 44, height: 44)
+    static let badge = CGSize(width: 15, height: 15)
+  }
+  
+  struct ImageColor {
+    static let avatar = UIColor.lightGray
+    static let badge = UIColor.red
+  }
+  
+  required init() {
+    super.init()
+    
+    avatarNode.backgroundColor = ImageColor.avatar
+    circularize(avatarNode, in: ImageSize.avatar)
+    
+    badgeNode.backgroundColor = ImageColor.badge
+    circularize(badgeNode, in: ImageSize.badge)
+    
+    makeSingleLine(for: usernameNode, with: "Hello world", fontSize: 17, textColor: .black)
+    makeSingleLine(for: subtitleNode, with: "This is a long long subtitle, with a long long appended string.", fontSize: 14, textColor: .lightGray)
+  }
+  
+  private func circularize(_ node: ASDisplayNode, in size: CGSize) {
+    assert(size.width == size.height)
+    node.style.preferredSize = size
+    node.cornerRadius = size.width / 2
```

Comment:
Not a blocker and don't spend too much time on this, but please consider other rounding techniques mentioned [here](http://texturegroup.org/docs/corner-rounding.html), if possible. I'm not concerned about the performance in this sample application, but rather using best practices in example code.